index.html

```diff
@@ -81,235 +81,215 @@
-        Document</a> contains a <a href=
-        "http://www.w3.org/TR/html5/infrastructure.html#valid-media-query">valid
-        media query</a> that <a href=
-        "http://www.w3.org/TR/html5/infrastructure.html#matches-the-environment">
-          matches the environment</a> of the user. Otherwise, if the media
-          query is invalid, or <a href=
-          "http://drafts.csswg.org/mediaqueries/#mq-list">evaluates to
-          false</a> [[!mediaqueries-4]], the user agent SHOULD NOT obtain the
-          resource.
+        <li>When the `media` attribute of the [link] element of a <a>preload
+        link</a> that is already [in a Document] contains a [valid media query
+        list] that [matches the environment] of the user. Otherwise, if the
+        media query is invalid, or [evaluates to false][] [[!mediaqueries-4]],
+        the user agent SHOULD NOT obtain the resource.
         </li>
       </ul>
-      <p>The user agent SHOULD abort the request if the `href` attribute on the
-      `link` element of an <a>preload link</a> is removed, or its value is set
-      to an empty string.</p>
-    </section>
-    <section>
-      <h2>Load and error events</h2>
-      <p>Once the attempt to obtain the resource is complete, the user agent
-      MUST, if the fetch was successful, queue a task to fire a simple event
-      named `load` at the `link` element, or, if the fetch failed to complete
-      for any reason, queue a task to fire a simple event named `error` at the
-      `link` element.</p>
-      <pre class="example highlight html">
-&lt;script&gt;
-  function preloadFinished(e) { ... }
-  function preloadError(e)  { ... }
-&lt;/script&gt;
-
-&lt;!-- listen for load and error events --&gt;
-&lt;link rel="preload" href="app.js" as="script" onload="preloadFinished()" onerror="preloadError()"&gt;
-</pre>
-      <div class="note">
-        The application can use the `load` event on a <a>preload</a> relation
-        as an indicator that the resource has been successfully fetched and is
-        now ready to be processed by the application - e.g. a stylesheet or
-        script can be applied to a document, and so on, without blocking on the
-        network.
-      </div>
-    </section>
-    <section>
-      <h2>Matching responses with requests</h2>
-      <p>Resources fetched via <code><a>preload</a></code> relation MUST be
-      retained by the user agent for the duration of the current navigation
-      until they are fetched with a matching request.</p>
-      <p>For example, if a JavaScript resource is fetched via a
+      <p>The user agent SHOULD abort the current request if the `href`
+      attribute of the [link] element of a <a>preload link</a> is changed,
+      removed, or its value is set to an empty string.</p>
+      <p>To <dfn>obtain the preload resource</dfn>, the user agent must run the
+      following steps:</p>
+      <ol>
+        <li>If the `href` attribute's value is the empty string, then abort
```

Comment:
> The destination of the link(s) is given by the href attribute, which must be present and must contain a valid non-empty URL potentially surrounded by spaces. If the href attribute is absent, then the element does not define a link ([source](https://html.spec.whatwg.org/multipage/semantics.html#link-type-stylesheet)).

If href is omitted then preload link is not defined. I think this is taken care of?